dotnet/src/Experimental/Process.IntegrationTests.Shared/ProcessMapTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0005 // Using directive is unnecessary.
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Process;
+using Xunit;
+#pragma warning restore IDE0005 // Using directive is unnecessary.
+
+namespace SemanticKernel.Process.IntegrationTests;
+
+/// <summary>
+/// Integration test focusing on cycles in a process.
+/// </summary>
+[Collection(nameof(ProcessTestGroup))]
+public class ProcessMapTests : IClassFixture<ProcessTestFixture>
+{
+    private readonly ProcessTestFixture _fixture;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ProcessCycleTests"/> class. This is called by the test framework.
+    /// </summary>
+    /// <param name="fixture"></param>
+    public ProcessMapTests(ProcessTestFixture fixture)
+    {
+        this._fixture = fixture;
+    }
+
+    /// <summary>
+    /// Tests a process which cycles a fixed number of times and then exits.
+    /// </summary>
+    /// <returns>A <see cref="Task"/></returns>
+    [Fact]
+    public async Task TestMapAsync()
+    {
+        // Arrange
+        ProcessBuilder process = new(nameof(TestMapAsync));
+
+        ProcessStepBuilder computeStep = process.AddStepFromType<ComputeStep>();
+        ProcessMapBuilder mapStep = process.AddMapForTarget(new ProcessFunctionTargetBuilder(computeStep));
```

Comment:
[brainstorming hypothetical scenario]
wondering if these 2 should be a single function? since compute is not actually used by the root process directly having it instantiated on the level computeStep could in theory also be used as a regular step:


```mermaid
graph TD
root-->map-->compute1["compute"]
root-->compute2["compute"]
```

Reply:
Good question.  Ben and I were discussing this also.  There is a case where an edge may be defined outside of the map-operation: https://github.com/microsoft/semantic-kernel/blob/processes-map-step/dotnet/src/Experimental/Process.UnitTests/Runtime.Local/LocalMapTests.cs#L295

```mermaid
graph TD
root["root process"]-->map
subgraph map
compute1["compute"]
end
compute1["compute"]--1-->orthogonal
compute1["compute"]--N-->union
root-->compute2["compute"]
```

Reply:
Worked with Ben to evolve what's going on here